src/runtime/vdom/vdom-render.ts

```diff
@@ -806,7 +811,10 @@ export const renderVdom = (hostRef: d.HostRef, renderFnResults: d.VNode | d.VNod
   const hostElm = hostRef.$hostElement$;
   const cmpMeta = hostRef.$cmpMeta$;
   const oldVNode: d.VNode = hostRef.$vnode$ || newVNode(null, null);
-  const rootVnode = isHost(renderFnResults) ? renderFnResults : h(null, null, renderFnResults as any);
+  const rootVnode: d.VNode =
+    !Array.isArray(renderFnResults) && isHost(renderFnResults)
```

Comment:
One thing we should verify - I'm oversimplifying the issue a little bit, but in TS it's possible to assign a property to an array and have the type checker be OK with it:
```ts
const foo: any = [];
foo.$tag$ = 'bar';
``` 

With our change to `isHost`, the previous implementation (of `isHost`) _could_ have returned `true` and narrowed the type of `renderFnResults` (since the original implementation was a type guard), even if `renderFnResults` was an array (assuming it had a `$tag$` property on it, which i believe only could happen in the situation I described above.

Can you do me a favor and verify the call chain to `renderVdom` to verify that something like this shouldn't happen? I'm not sure if there's any type erasure or widening in that chain that could lead to an issue here, but probably best be safe than sorry.

Reply:
It's hard to say 100% because unfortunately we more or less have typechecking turned off in `update-component.ts` (lots of `any` scattered around 😞) but looking through the call sites for `renderVdom`, `callRender`, and `updateComponent` I don't see anything like that going on. The reason I opened this PR in the first place was in part that I get a little scared of how un-checked some of these files are...

Reply:
Fair, so long as we're fairly confident here. Worst case scenario, we need to revert this (and if there is a rendering bug, we know where to look first)